doc/compiler/tril/TrilLanguageReference.md

```diff
@@ -0,0 +1,194 @@
+# Tril Language Reference
+
+In may ways, the Tril language mirrors the Testarossa Intermediate Language,
+_Trees_. 
+
+This language reference intends to document mostly the places where special
+behaviour has been created in Tril to make writing test cases easier. 
+
+# Basics
+
+Tril is built out of two main components: 
+
+1. A very permissive S-expression parser.
+2. An IlGenerator that walks the AST produced by said parser, and generates
+   trees. 
+
+# Syntax
+
+Tril represents Trees using S-expressions, with key-value properties to augment
+a particular node with more information. 
+
+* `(x (y) (z))` represents some `x` with `y` and `z` as children.
+* `(x foo=bar baz=braz (y) (z)` is the same tree as above, but where the node
+  for `x` is augmented with two properties with values. 
+
+  Anything may be provided as a property for a node, however, unknown properties will be silently ignored.
+
+
+## Special Forms: 
+
+The key to Tril is the unlocking of extra behaviour using 'special forms', or
+atoms that represent certain behaviours in Testarossa. 
+
+### Methods
+
+A whole method in Tril is wrapped in a `method` expression. 
+
+#### Properties 
+
+* `return` _Mandatory_ is the return type of the method. 
+* `args` _Optional_ is a square-bracketed list of arguments representing the
+  types of the parameters, and the order in which they appear. 
+
+  eg. `args=[Int32,Double,Float]`
+    
+
+
+### Block
+
+In the Testarossa IL, all trees exist within the context of a basic block. 
```

Comment:
Maybe mention that blocks _must_ children of a method (otherwise, they will not be recognized as blocks).

Reply:
Also maybe mention that all child expressions of a block are interpreted as opcodes.